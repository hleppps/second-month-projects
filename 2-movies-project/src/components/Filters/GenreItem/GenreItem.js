import { Form } from "react-bootstrap";

const GenreItem = ({ isSelected, ...props }) => {
  const onCheckHandler = () => {
    props.onSelect(!isSelected, props.id);
  };

  return (
    <Form.Check
      data-testid="checkbox"
      id={props.id}
      type="checkbox"
      label={props.name}
      checked={isSelected}
      onChange={onCheckHandler}
    />
  );
};

export default GenreItem;
