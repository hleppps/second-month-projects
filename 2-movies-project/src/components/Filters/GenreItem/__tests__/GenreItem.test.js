import { render, cleanup, fireEvent, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import GenreItem from "../GenreItem";

afterEach(cleanup);

describe("GenreItem component", () => {
  it("renders checkbox with correct label", () => {
    const label = "Label Name";

    render(<GenreItem name={label} />);

    expect(screen.getByText(label)).toBeInTheDocument();
  });

  it("calls a function when checked", () => {
    const onSelect = jest.fn();

    const { getByTestId } = render(<GenreItem onSelect={onSelect} />);

    fireEvent.click(getByTestId("checkbox"));

    expect(onSelect).toBeCalled();
  });

  it("matches a snapshot", () => {
  const { asFragment } = render(<GenreItem />);
  expect(asFragment(<GenreItem />)).toMatchSnapshot();
  });
});
