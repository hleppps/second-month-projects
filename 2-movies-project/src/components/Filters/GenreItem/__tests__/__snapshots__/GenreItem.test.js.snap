// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenreItem component matches a snapshot 1`] = `
<DocumentFragment>
  <div
    class=""
  >
    <input
      class="form-check-input"
      data-testid="checkbox"
      type="checkbox"
    />
  </div>
</DocumentFragment>
`;
